tests/installation/registration/registration_with_regcode.pm

```diff
@@ -0,0 +1,26 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Register the system with scc in the installer with registration code.
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi qw(check_var get_var send_key save_screenshot);
+use registration qw(handle_scc_popups process_modules skip_registration);
+
+sub run {
+    my $reg_code = get_var('SCC_REGCODE');
+    $testapi::distri->get_registration()->register_product_with_regcode($reg_code);
+    handle_scc_popups();
```

Comment:
Temporary solution. With theis we keep the old way of deciding whether we we use or not the online repositories, and it deals with the intermediate popups. Or do we want to deal with these pop-ups with libyui ? with strict expectations, we should only have to deal with this piece below as the rest is for SLEl 12:

            while ($counter--) {
            die 'Registration repeated too much. Check if SCC is down.' if ($counter eq 1);
            assert_screen(\@tags, timeout => 360);
            if (match_has_tag('import-untrusted-gpg-key')) {
                handle_untrusted_gpg_key;
                next;
            }
            elsif (match_has_tag('nvidia-validation-failed')) {
                # sometimes nvidia driver repos are unreliable
                send_key 'alt-y';
                record_soft_failure 'bsc#1144831';
                next;
            }
            elsif (match_has_tag('contacting-registration-server')) {
                # sometimes SCC just takes its time - just continue looking after a while
                sleep 5;
                next;
            }
            elsif (match_has_tag('refreshing-repository')) {
                # it takes some time to refresh repos
                sleep 5;
                next;
            }

Reply:
We should definitely get rid of approach with `match_has_tag` in our new test modules, as it makes test unpredictable. If we would do automatic installation but not a test, it may be useful to make the program to be very smart and decide what to do with some particular popup, which appear. But I guess we discussed that in a team and agreed that test should be dumb and perform expected steps. 

If some popup should be handled in test suite, it should not be some random thing, it is a separate test and probably we need to create a separate test module for this case.

So, for new tests we should define if some popups should be handled and handle them in separate test module. For `scc_registration` keep it as is for now, as in your task other job groups are out of scope and we cannot say right now if some popups need to be handled there.

Reply:
I agree, but what to do with soft-failures ?

Reply:
eg "# sometimes nvidia driver repos are unreliable"

Reply:
Do the issues happen? In case it is really happen and it is something sporadic, leave the condition for this particular issue for now and create a ticket to investigate how can we improve our test module on that.

But I guess there should be no issues in our job group. At least I do not remember them.

Reply:
For example in functional group, they are not even using yaml scheduling (eg here https://openqa.suse.de/tests/6599569#settings). I'd rather not use libyui in that module as we'd need to find every single test suite that uses it, create tons of yaml schedules, or mess with main.pm...